tests/kafkatest/tests/core/zookeeper_migration_test.py

```diff
@@ -86,10 +87,35 @@ def do_migration(self, roll_controller = False, downgrade_to_zk = False):
                 controller.stop_node(node)
                 controller.start_node(node)
 
+        if downgrade_to_zk:
+            self.logger.info("Shutdown brokers to avoid waiting on unclean shutdown")
+            for node in self.kafka.nodes:
+                self.kafka.stop_node(node)
+                metadata_log_dir = KafkaService.METADATA_LOG_DIR + "/__cluster_metadata-0"
+                for node in self.kafka.nodes:
+                    assert path_exists(node, metadata_log_dir), "Should still have a metadata log on the brokers."
+
+            self.logger.info("Shutdown KRaft quorum")
+            for node in controller.nodes:
+                controller.stop_node(node)
+
+            self.logger.info("Deleting controller ZNode")
+            self.zk.delete(path="/controller", recursive=True)
+
+            self.logger.info("Rolling brokers back to ZK mode")
+            self.kafka.downgrade_kraft_broker_to_zk(controller)
+            for node in self.kafka.nodes:
+                self.kafka.start_node(node)
+
+            # This blocks until all brokers have a full ISR
+            self.wait_until_rejoin()
```

Comment:
The test name is `test_online_migration` but we're shutting every broker down to apply the downgrade, it seems like a contradiction? Does the downgrade require a full shutdown of the cluster?